OJS.Workers.SubmissionProcessors/SubmissionProcessor.cs

```diff
@@ -131,11 +131,52 @@ private IOjsSubmission GetSubmissionForProcessing()
 
                 try
                 {
-                    if (!this.submissionsFilteringService.CanProcessSubmission(submission, this.SubmissionWorker))
+                    var workerStateForSubmission =
+                        this.submissionsFilteringService.GetWorkerStateForSubmission(submission, this.SubmissionWorker);
+
+                    if (workerStateForSubmission == WorkerStateForSubmission.Ready)
+                    {
+                        return submission;
+                    }
+
+                    if (workerStateForSubmission == WorkerStateForSubmission.Unhealthy)
                     {
                         this.SubmissionProcessingStrategy.ReleaseSubmission();
+                        this.Logger.Error($"Submission with Id: {submission.Id}, cannot be processed. Reason: Cannot be processed by the worker.");
                         return null;
                     }
+
+                    var message = string.Empty;
+                    switch (workerStateForSubmission)
+                    {
+                        case WorkerStateForSubmission.DisabledStrategy:
+                            message = "Strategy is disabled.";
+                            break;
+                        case WorkerStateForSubmission.NotEnabledStrategy:
+                            message = "Strategy is not enabled.";
+                            break;
+                        case WorkerStateForSubmission.NullableSubmission:
+                            message = "Submission is null";
+                            break;
+                        case WorkerStateForSubmission.DisabledCompilerType:
+                            message = "Compiler type is disabled.";
+                            break;
+                        default:
+                            throw new ArgumentOutOfRangeException(
+                                $"Worker state for submission: {workerStateForSubmission} is invalid.");
+                    }
+
+                    if (workerStateForSubmission != WorkerStateForSubmission.NullableSubmission)
+                    {
+                        submission.CompilerComment =
+                            "The submission cannot be handled due to unsupported submission type. Please contact an Administrator.";
+                    }
+
+                    this.Logger.Error($"Submission with Id: {submission.Id}, cannot be processed. Reason: {message} ");
```

Comment:
Make sure this or similar message (with Reason included) is visible by admins in the submission details as well.

Reply:
Under the logging, there is a call of OnError method, which will set the message for the admin